### ISSUE TITLE
 Gradle files unchanged after linking

### ISSUE THREAD

**Issue by joshg111** [rnpm/rnpm#195]
After linking with: "rnpm link " I see added to MainActivity.java, but I don't see the necessary changed to build.gradle or settings.gradle. My apps package.json file is below. The output of the link command is successful. Also, this might be a related issue, but when i run "rnpm install " I get the following output:
`rnpm-link info Module react-native-fbsdk has been successfully installed & linked`

But in this case the linking doesn't work at all. Seems to ignore the MainActivity and gradle files, where at least the "rnpm link " makes the changes to MainActivity.

Please let me know what other info you need.

```
{
  "name": "event_photo",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "alt": "^0.18.4",
    "native-base": "^0.4.7",
    "react": "15.0.2",
    "react-native": "^0.26.3",
    "react-native-contacts": "^0.2.8",
    "react-native-fbsdk": "^0.2.2",
    "react-native-image-picker": "^0.19.4",
    "react-native-router-flux": "^3.31.0",
    "react-native-vector-icons": "^2.0.3"
  }
}


```

**Comment by joshg111** [comment]
I'm trying to link react-native-fbsdk. I can do it manually, but I seem to continuously have this problem even with other packages.

**Comment by Kureev** [comment]
That's quite strange. Just tried to generate a blank project using react@0.26.3 and it worked. Did you apply any changes to your gradle files?

**Comment by joshg111** [comment]
Yes, there are many other changes to the gradle file. Actually my project was originally a non react native android project, then it evolved into react native. During the evolution there were a number of issue I ran into. Although it has been working fine for a while now, I'm thinking of remaking the project from scratch. Perhaps this cleanup will get rid of this strange problem. Will update the thread in the next few days with the result.

**Comment by Kureev** [comment]
it's sensitive to indentation in gradle files, maybe you changed it for key-lines (we use find-replace approach)

**Comment by joshg111** [comment]
After creating a fresh react native project all the android files seem to be updated accordingly when running rnpm link with the exception of the MainApplication.java file. Rnpm should be adding the package names in the array defined in getPackages overriden function, but I end up having to do it manually with each package I install. 

Also, I find that rnpm install does not install the package or link the package, but it gives the following output:
`rnpm-link info Module react-native-fbsdk has been successfully installed & linked`

**Comment by Kureev** [comment]
Thanks for your feedback, @joshg111!

We fixed MainApplication patch issue in react-native (rnpm is going to be deprecated soon as we merged it into react-native core). About fbsdk: it's a very tricky module. I think it'll be great if you can join our discussion here https://github.com/facebook/react-native/issues/8623.
